### PR TITLE
Notification permission implementation

### PR DESCRIPTION
**Pull request by ssniks**

### CODE REVIEW

File: androidApp/src/main/java/app/suprsend/android/WelcomeActivity.kt [hunk 1]
```diff
@@ -70,6 +76,33 @@ class WelcomeActivity : AppCompatActivity() {
...
         subscribeToTopic()
+        requestNotificationPermission(NOTIFICATION_PERMISSION_REQUEST_CODE)
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == NOTIFICATION_PERMISSION_REQUEST_CODE && !isNotificationPermissionGranted()) {
+            // You can show a dialog which explains the intent of this permission request how it is important
+            // for certain features to work and re-request the permission by calling requestNotificationPermission()
+            AlertDialog.Builder(this)
```
Comment (ssniks) [inline androidApp/src/main/java/app/suprsend/android/WelcomeActivity.kt:87]: needs testing

File: library/src/main/java/app/suprsend/notification/NotificationPermissionHelper.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    fun Activity.requestNotificationPermission(
+        requestCode: Int,
+        snackBarDescription: String = "Notification is blocked",
+        snackBarActionTitle: String = "Settings"
+    ) {
+        val notificationManagerCompat = NotificationManagerCompat.from(this)
+        Log.i("permission", "areNotificationsEnabled : ${notificationManagerCompat.areNotificationsEnabled()}")
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```
Comment (ssniks) [inline library/src/main/java/app/suprsend/notification/NotificationPermissionHelper.kt:27]: Test on device lower than m

File: library/src/main/java/app/suprsend/notification/SSNotificationHelper.kt [hunk 3]
```diff
@@ -78,6 +79,13 @@ object SSNotificationHelper {
     private fun showRawNotification(context: Context, rawNotification: RawNotification, pushVendor: String? = null) {
         try {
             Logger.i("notification","showRawNotification $rawNotification")
+
+            val notificationManagerCompat = NotificationManagerCompat.from(context)
+            if(!notificationManagerCompat.areNotificationsEnabled()){
+                Logger.e("notification","Notifications are disabled please request the Manifest.permission.POST_NOTIFICATIONS permission")
```
Comment (ssniks) [inline library/src/main/java/app/suprsend/notification/SSNotificationHelper.kt:85]: We can implement log listener

File: androidApp/src/main/java/app/suprsend/android/WelcomeActivity.kt [hunk 4]
```diff
@@ -70,6 +76,33 @@ class WelcomeActivity : AppCompatActivity() {
...
+                }
+                .setNegativeButton("Deny") { dialogInterface, p1 ->
+                }.show()
+
+
+        }
+    }
+
+
+    private fun isNotificationPermissionGranted(): Boolean {
```
Comment (ssniks) [inline androidApp/src/main/java/app/suprsend/android/WelcomeActivity.kt:100]: move this to utility

**Review by sivaram000 — APPROVED** [review]

**Pull request merged**